### DISCUSSION TITLE
helm-controller: Helm charts from git

### DISCUSSION THREAD

**Discussion by carlpett** [fluxcd/flux2#75] [Proposals] [answered]
Hey,
After reading about https://github.com/fluxcd/helm-controller/issues/4 and discussions in Slack, I saw this in the roadmap non-goals:
https://github.com/fluxcd/toolkit/blob/b7727e2659d2476312abdb43f7c9df0f64bfa0e0/docs/roadmap/index.md#L72

I did not find the underlying reason for this decision documented, and would like to understand why this is a non-goal? From my perspective it would be unfortunate as it is a workflow that has been very useful! What are the issues that makes it unsuitable to keep?

**Comment by hiddeco** [comment]
This "non-goal" should be read as "we are not planning to include this in the first GA release". I understand the value of the feature itself, and would even like to see it as a "stretch goal", hence https://github.com/fluxcd/source-controller/issues/56.

The reason (for me) that it is marked as a non-goal for the GA is that the Helm Operator was developed from a Git perspective which in the end resulted in an insane amount of complexity within the operator because of various tricks we had to pull off. By targeting "native Helm" functionalities first I am hoping to prevent this as the foundation/design is then tailor made for how Helm likes to behave.

**Comment by carlpett** [comment]
Ah, language confusion on my part, sorry about that! But that is reassuring to hear, and sounds like a good reason for not going ahead with it as the first release.
I haven't read the proposals in detail yet, so apologies if this is covered already... But would it make sense to look at a sidecar approach with a "git syncer" sidecar dealing with the generic task of updating repos, a shared volume, and the new Helm controller could work with files on disk (in addition to Helm repos)?
Or maybe the syncronisation between the two would add sufficient complexity that it is not a net win?

**Comment by hiddeco** [comment, marked as answer]
The idea is to make the source-controller capable of producing `HelmChart` artifacts from a `GitRepository` (as an extension to already implemented `HelmRepository` -> `HelmChart`). This way the helm-controller only needs to know how to work with a single resource type (`HelmChart`), which solves the complexity issue.

**Comment by carlpett** [comment]
Ah, cool! I should go read the proposals :) But then I think my concerns are very well met, thanks for explaining!
